### PR TITLE
Clarify podman machine volume mounting behavior under WSL

### PR DESCRIPTION
**Pull request by wbrefvem**
#### Does this PR introduce a user-facing change?

No

Addresses #23763 

AFAICT the help text piece is addressed in https://github.com/containers/common/pull/2100.

### CODE REVIEW

**Comment by mheon** [comment]
LGTM
@baude PTAL

File: docs/source/markdown/podman-machine-init.1.md.in [hunk 1]
```diff
@@ -160,6 +160,11 @@ Example: `-v "$HOME/git:$HOME/git:ro,security_model=none"`
 Default volume mounts are defined in *containers.conf*.  Unless changed, the default values
 is `$HOME:$HOME`.
 
+**Note on Windows Subsystem for Linux (WSL)**
+Since the entire `C:` drive is mounted at `/mnt/c` by default in WSL, passing
```
Comment (Luap99) [inline docs/source/markdown/podman-machine-init.1.md.in:164]: AFAIK, this is not just restricted to the C drive, is it? Doesn't windows mount all drives that way, e.g. /mnt/d and so on? Maybe add this as well in the note here

Comment (wbrefvem) [reply docs/source/markdown/podman-machine-init.1.md.in:164]: Good point, it does seem that that's the expected behavior. I'll make the update.

**Review by Luap99 — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
/lgtm
Thanks @wbrefvem

**Pull request merged**